cg/meta/demultiplex/utils.py

```diff
@@ -209,14 +211,15 @@ def is_file_relevant_for_demultiplexing(file: Path) -> bool:
 def is_syncing_complete(source_directory: Path, target_directory: Path) -> bool:
     """Returns whether all relevant files for demultiplexing have been synced from the source to
     the target."""
-    manifest_file = Path(source_directory, DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST)
-    if not manifest_file.exists():
-        LOG.debug(
-            f"{source_directory} does not contain a "
-            f"{DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST} file. Skipping."
-        )
+    manifest_files: list[Path] = [
+        Path(source_directory, DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST),
+        Path(source_directory, DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST),
+    ]
+    existing_files: list[Path] = [file for file in manifest_files if file.exists()]
+    if not existing_files:
+        LOG.debug(f"{source_directory} does not contain a manifest file. Skipping.")
         return False
-    files_at_source: list[Path] = parse_manifest_file(manifest_file)
+    files_at_source: list[Path] = parse_manifest_file(existing_files[0])
```

Comment:
Not super happy with this block. Is it clear what is done here?

Reply:
so basically there will never be the case in which the two manifests exist because one is for 6000 and the other one is for X, right? So you pick the first one because the list will always have one item

Reply:
It is true that there should not be two such files, but even if there are, any of them would be fine to parse.

Reply:
Slightly out of scope, but the function could be split into smaller ones like this for example
```python
from typing import Optional

def get_existing_manifest_file(source_directory: Path) -> Optional[Path]:
    """Returns the first existing manifest file in the source directory."""
    manifest_file_paths = [
        Path(source_directory, DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST),
        Path(source_directory, DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST),
    ]
    for file_path in manifest_file_paths:
        if file_path.exists():
            return file_path

def are_all_files_synced(files_at_source: list[Path], target_directory: Path) -> bool:
    """Checks if all relevant files in the source are present in the target directory."""
    for file in files_at_source:
        if is_file_relevant_for_demultiplexing(file) and not Path(target_directory, file).exists():
            LOG.info(
                f"File: {file}, has not been transferred from source to {target_directory}"
            )
            return False
    return True

def is_syncing_complete(source_directory: Path, target_directory: Path) -> bool:
    """Returns whether all relevant files for demultiplexing have been synced from the source to the target."""
    
    existing_manifest_file = get_existing_manifest_file(source_directory)
    
    if existing_manifest_file is None:
        LOG.debug(f"{source_directory} does not contain a manifest file. Skipping.")
        return False

    files_at_source = parse_manifest_file(existing_manifest_file)
    return are_all_files_synced(files_at_source, target_directory)

```

Reply:
I think a way of making this less complicated is to sub-hire a function that extracts **the** existing file from a list of files and returns `None` if neither of the files exists. I think you could leave it as it is if you think this change is out of scope.

Reply:
Implemented @seallard's and @diitaz93's suggestion